### ISSUE TITLE
Need method "needsPartFile()" implemented in Storage, for a External Storage

### ISSUE THREAD

**Issue by viniciuscb** [nextcloud/server#3420]
Hi. I am creating an external storage that does not accept part file. When making the method OCA\DAV\Connector\Sabre\File->needsPartFile($storage) return false for my storage, the file upload works as expected. I've done it already in my fork, creating the method needsPartFile in the OCP\Files\Storage interface, but I don't know how to execute the auto-tests, so I did not submit as a pull request.

Can you solve this, allowing custom storages specify this "configuration" (needsPartFie)? Or guide me through the testing procedure so I can contribute with pull request. Thank you.

 
### Server configuration
**Operating system**: Linux

**Web server:** Apache

**Database:**  SQLite

**PHP version:** 7

**Nextcloud version:** (see Nextcloud admin page) 11.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh

**Where did you install Nextcloud from:** git

**Signing status:**
<details>
<summary>Signing status</summary>

```
Integrity checker has been disabled. Integrity cannot be verified.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - comments: 1.1.0
  - dav: 1.1.1
  - federatedfilesharing: 1.1.1
  - federation: 1.1.1
  - files: 1.6.1
  - files_external: 1.1.2
  - files_external_ftp: 0.1.0
  - files_sharing: 1.1.1
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - lookup_server_connector: 1.0.0
  - matrixbridge: 0.0.6
  - provisioning_api: 1.1.0
  - sharebymail: 1.0.1
  - systemtags: 1.1.3
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - user_ldap: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - admin_audit
  - encryption
  - testing
  - updatenotification
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "oc20kc32ygj2",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud11.localhost"
        ],
        "datadirectory": "\/srv\/www\/htdocs\/nextcloud11\/data",
        "overwrite.cli.url": "http:\/\/nextcloud11.localhost",
        "dbtype": "sqlite3",
        "version": "11.0.1.2",
        "logtimezone": "UTC",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "loglevel": 0
    },
    "apps": {
        "backgroundjob": {
            "lastjob": "6"
        },
        "comments": {
            "installed_version": "1.1.0",
            "types": "logging",
            "enabled": "yes"
        },
        "core": {
            "installedat": "1486573913.671",
            "lastupdatedat": "1486576062",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "public_files": "files_sharing\/public.php",
            "oc.integritycheck.checker": "[]",
            "lastcron": "1486668302"
        },
        "dav": {
            "installed_version": "1.1.1",
            "types": "filesystem",
            "enabled": "yes"
        },
        "federatedfilesharing": {
            "installed_version": "1.1.1",
            "types": "",
            "enabled": "yes"
        },
        "federation": {
            "installed_version": "1.1.1",
            "types": "authentication",
            "enabled": "yes"
        },
        "files": {
            "installed_version": "1.6.1",
            "types": "filesystem",
            "enabled": "yes",
            "cronjob_scan_files": "500"
        },
        "files_external": {
            "enabled": "yes",
            "installed_version": "1.1.2",
            "types": "filesystem"
        },
        "files_external_ftp": {
            "enabled": "yes",
            "installed_version": "0.1.0",
            "types": "filesystem"
        },
        "files_sharing": {
            "installed_version": "1.1.1",
            "types": "filesystem",
            "enabled": "yes"
        },
        "files_trashbin": {
            "installed_version": "1.1.0",
            "types": "filesystem",
            "enabled": "yes"
        },
        "files_versions": {
            "installed_version": "1.4.0",
            "types": "filesystem",
            "enabled": "yes"
        },
        "lookup_server_connector": {
            "installed_version": "1.0.0",
            "types": "authentication",
            "enabled": "yes"
        },
        "matrixbridge": {
            "enabled": "yes",
            "installed_version": "0.0.6",
            "types": "filesystem"
        },
        "provisioning_api": {
            "installed_version": "1.1.0",
            "types": "prevent_group_restriction",
            "enabled": "yes"
        },
        "sharebymail": {
            "installed_version": "1.0.1",
            "types": "filesystem",
            "enabled": "yes"
        },
        "systemtags": {
            "installed_version": "1.1.3",
            "types": "logging",
            "enabled": "yes"
        },
        "theming": {
            "installed_version": "1.1.1",
            "types": "logging",
            "enabled": "yes"
        },
        "twofactor_backupcodes": {
            "installed_version": "1.0.0",
            "types": "",
            "enabled": "yes"
        },
        "updatenotification": {
            "installed_version": "1.1.1",
            "types": "",
            "enabled": "no"
        },
        "user_ldap": {
            "enabled": "yes",
            "installed_version": "1.1.1",
            "types": "authentication"
        },
        "workflowengine": {
            "installed_version": "1.1.1",
            "types": "filesystem",
            "enabled": "yes"
        }
    }
```
</details>

**Are you using external storage, if yes which one:**  https://gitlab.com/eita/matrixbridge

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** 

### Client configuration
**Browser:** Chromium

**Operating system:** Linux

**Comment by schiessle** [comment]
Did you already looked here: https://docs.nextcloud.com/server/11/developer_manual/core/unit-testing.html

Most distribution already offer packages for phpunit. You just need to install them and then run `./autotest sqlite` (sqlite is the easiest one because you don't need to setup a database).

Would be great if you could propose a pull request. Don't worry if it isn't perfect right from the beginning. If we have a pull request we can help you improving it and our test server will also run the tests for you. Not perfect (because you have to wait quite some time) but better then nothing until you managed to set it up locally.

**Comment by viniciuscb** [comment]
Here it is.
https://github.com/nextcloud/server/pull/3600

**Comment by MorrisJobke** [comment]
Fixed by #3600
